### PR TITLE
create a SimpleSpanBatchSender and wire up to the example

### PR DESCRIPTION
**Pull request by breedx-nr**
This addresses #58.

A few other notable changes:
* adds builder verification for the attributes json instance being supplied (wasn't checked before)
* adds test coverage around the builder verifications

I'm also looking for feedback on the fact that I exposed 4 methods on the `SimpleSpanBatchSender` -- 2 that return the builder in case the user wants additional customization (like in the example) and 2 that return the 'Sender directly.

### CODE REVIEW

File: telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSenderBuilder.java [hunk 1]
```diff
@@ -35,24 +35,28 @@ public MetricBatchSender build() {
...
     Utils.verifyNonNull(attributesJson, "an AttributesJson implementation is required.");
 
-    if (metricsUrl == null) {
-      try {
-        metricsUrl = constructMetricsUrlWithHost(URI.create("https://metric-api.newrelic.com/"));
-      } catch (MalformedURLException e) {
-        throw new UncheckedIOException("Bad hardcoded URL", e);
-      }
-    }
+    URL url = getOrDefaultMetricsUrl();
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSenderBuilder.java]: I like it

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  public static SpanBatchSenderBuilder builder(String apiKey) {
+    return builder(apiKey);
+  }
+
+  public static SpanBatchSender build(String apiKey, Duration callTimeout) {
+    return builder(apiKey, callTimeout).build();
+  }
+
+  public static SpanBatchSenderBuilder builder(String apiKey, Duration callTimeout) {
```
Comment (jkwatson) [inline telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: I'd like to change the metric version of this to look more like this, as well.

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.google.gson.GsonBuilder;
+import com.newrelic.telemetry.http.HttpPoster;
+import com.newrelic.telemetry.spans.SpanBatchSender;
+import com.newrelic.telemetry.spans.SpanBatchSenderBuilder;
+import java.time.Duration;
+
+public class SimpleSpanBatchSender {
+
+  public static SpanBatchSender build(String apiKey) {
+    return build(apiKey, Duration.ZERO);
```
Comment (jkwatson) [inline telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: I don't think ZERO is a good value here. We should choose a good default. ZERO means infinite timeout, and we've gotten feedback that this is a bad choice.

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    return builder(apiKey);
+  }
+
+  public static SpanBatchSender build(String apiKey, Duration callTimeout) {
+    return builder(apiKey, callTimeout).build();
+  }
+
+  public static SpanBatchSenderBuilder builder(String apiKey, Duration callTimeout) {
+    HttpPoster http = new OkHttpPoster(callTimeout);
+    Gson gson = new GsonBuilder().create();
```
Comment (jkwatson) [inline telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: does this do anything different than just `new Gson()` ?

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java (hunk 3 above)
Comment (breedx-nr) [reply telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: Heh, yeah, I was borrowing heavily from the other simple sender.  I agree tho and will change it to 2 seconds as a reasonable starting point.

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java (hunk 4 above)
Comment (breedx-nr) [reply telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: The docs say
```
 * <p>Use this builder to construct a {@link Gson} instance when you need to set configuration
 * options other than the default. For {@link Gson} with default configuration, it is simpler to
 * use {@code new Gson()}. {@code GsonBuilder} is best used by creating it, and then invoking its
 * various configuration methods, and finally calling create.</p>
 *
```
so yeah, I'll change it to the simpler constructor.  Good catch!  🏆

File: telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java (hunk 2 above)
Comment (breedx-nr) [reply telemetry_components/src/main/java/com/newrelic/telemetry/SimpleSpanBatchSender.java]: I picked that up in this PR now too.

**Comment by breedx-nr** [comment]
Ready for another looksie.

**Comment by jkwatson** [comment]
let's ship it. you need to rebase, and we need to write about a million lines of javadoc.

**Pull request merged**